docs/api/commands/each.mdx

```diff
@@ -116,8 +116,7 @@ function.
 
 ### Assertions [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Assertions)
 
-- `.each()` will only run assertions you have chained once, and will not
-  [retry](/guides/core-concepts/retry-ability).
+- `.each()` has no implicit assertions.
```

Comment:
I only know what this means because I have review our code so often. I do not believe our users will understand what this truly means. We need to to provide more context here.

I think only calling out when a Cypress command performs an internal assertion makes the most sense.

Also when I read these docs, I am more interested in understanding how assertions or if assertion retry for a command chain. In the case of `.each()`, assertions are not retried because .each() is not a query command. It's yielded value is static so re-trying the assertions will not change the result.

Something like this might be better:
```
- When the `.each()` command runs, Cypress performs chained assertions once. because `cy.exec()` is not a query and the
  [only executes the command once](/guides/core-concepts/retry-ability#Only-queries-are-retried), the yielded value is static so retrying the assertion does not change the assertion result.
```

Reply:
Right - this is changing so assertions always retry - however this isn't clear how it ties to the subject being "updated" with queries vs non-query commands.

Reply:
Hm. I didn't like this at first, but the more I read it, the more I think you're right, and reitterating how queries / non-queries interact in the `### Assertions` section of each page is the right approach.

Reply:
Have done this in subsequent revisions of the PR; going to mark this conversation resolved, and we can discuss more on a fresh version of the copy if needed.